### ISSUE TITLE
Graceful shutdown - fastify.close is not a function

### ISSUE THREAD

**Issue by silversoul93** [mia-platform/lc39#308]
## The description of the bug or the rationale of your proposal

Hi guys,
we have a Node.JS microservice that uses the Mia-Platform [_custom-plugin-lib_](https://github.com/mia-platform/custom-plugin-lib) with Fastify.

We noticed a weird behaviour when the service was shut down by Kubernetes because of multiple `livenessProbe` failures.
When multiple `livenessProbe` fail, Kubernetes send a `SIGTERM` to the pod that performs a graceful shutdown, calling the `fastify.close()` function. Despite the `.close` function exists, the last log of the service is the following one:
```
{"level":50,"time":1696341835306,"pid":7,"hostname":"redirect-manager-64b966f9df-96598","debugInfo":{"status":"KO","name":"redirect-manager","version":"3.3.0-rc.1"},"msg":"Service status is not ok"}
/home/node/app/node_modules/@mia-platform/lc39/lib/launch-fastify.js:172
      fastify.close()
              ^

TypeError: fastify.close is not a function
    at Timeout._onTimeout (/home/node/app/node_modules/@mia-platform/lc39/lib/launch-fastify.js:172:15)
    at listOnTimeout (node:internal/timers:569:17)
    at process.processTimers (node:internal/timers:512:7)
```

We are using Node.JS `v18.18.0` and the version v6.0.0 of the `custom-plugin-lib`.

Did you ever have this error?

Thanks.

## A snippet of code for replicating the issue or showing the proposal usage if applicable
```javascript
module.exports = customService(async function index(service) {
  ...
  service.addHook('onClose', onClose)
}

async function onClose(service, done) {
  const { redirectionCompletionListener } = service
  await redirectionCompletionListener.stopListen()
  done()
} 
```

## The expected result for your bug

no crash

### Your environment

Node.JS `v18.18.0`
custom-plugin-lib `v6.0.0`

**Comment by davidebianchi** [comment]
Hi @silversoul93! I opened a PR that should resolve this issue. Anyhow, keep attention to not mix async function and callback params (as the done params).

For further information, you can check the fastify docs: https://fastify.dev/docs/latest/Reference/Hooks/#onclose

**Comment by davidebianchi** [comment]
Fix released in 7.0.2
